### PR TITLE
WT-9714 Ensure that call-log generated by python tests runs on the simulator.

### PR DESCRIPTION
**Pull request by Siddhartha8899**
A few issues were identified when passing the call log generated from the python tests to the simulator. This PR focuses on resolving the same.

- The configs can have spaces. For instance, `oldest_timestamp    =          2 ,     stable   = 15` is a valid config.
- `set_timestamp` takes a config `force` that skips the validation of timestamps.
- Improved the JSON output.

### CODE REVIEW

File: test/simulator/timestamp/src/connection_simulator.cpp
Comment (etienneptl) [inline test/simulator/timestamp/src/connection_simulator.cpp:173]: Should we explicitly set `force` to `false` in an `else` statement?

**Review by etienneptl — APPROVED** [review]
LGTM with one np

File: test/simulator/timestamp/src/connection_simulator.cpp
Comment (Siddhartha8899) [reply test/simulator/timestamp/src/connection_simulator.cpp:173]: It is set to false [here](https://github.com/wiredtiger/wiredtiger/pull/8186/files#diff-402b40f7efe9213e945ecef68f5de126368d636319b6456a35145ff1c4725b00R191)

File: test/simulator/timestamp/call_log_manager/call_log_manager.cpp
Comment (etienneptl) [inline test/simulator/timestamp/call_log_manager/call_log_manager.cpp:42]: In terms of error handling, is `exit` ok? You may want to return an exception?

File: test/simulator/timestamp/call_log_manager/call_log_manager.cpp
Comment (etienneptl) [inline test/simulator/timestamp/call_log_manager/call_log_manager.cpp:50]: ditto

File: test/simulator/timestamp/call_log_manager/call_log_manager.cpp
Comment (etienneptl) [inline test/simulator/timestamp/call_log_manager/call_log_manager.cpp:56]: I would probably do a while just in case. Same for the comma.

File: test/simulator/timestamp/call_log_manager/call_log_manager.cpp
Comment (etienneptl) [inline test/simulator/timestamp/call_log_manager/call_log_manager.cpp:60]: Should we check if the first character is `[`? Same for the last character, if it's `]`? It could be overkill.

File: test/simulator/timestamp/call_log_manager/call_log_manager.cpp
Comment (Siddhartha8899) [reply test/simulator/timestamp/call_log_manager/call_log_manager.cpp]: Updated in [b8e864b](https://github.com/wiredtiger/wiredtiger/pull/8186/commits/b8e864b9d1665b5de62f173504da094f7402a400)

File: test/simulator/timestamp/call_log_manager/call_log_manager.cpp
Comment (Siddhartha8899) [reply test/simulator/timestamp/call_log_manager/call_log_manager.cpp]: Updated in [b8e864b](https://github.com/wiredtiger/wiredtiger/pull/8186/commits/b8e864b9d1665b5de62f173504da094f7402a400)

File: test/simulator/timestamp/call_log_manager/call_log_manager.cpp
Comment (Siddhartha8899) [reply test/simulator/timestamp/call_log_manager/call_log_manager.cpp]: Updated in [76ee71a](https://github.com/wiredtiger/wiredtiger/pull/8186/commits/76ee71a2f16b8cedf531ca0fa2b1820284311c3c)

File: test/simulator/timestamp/call_log_manager/call_log_manager.cpp
Comment (Siddhartha8899) [reply test/simulator/timestamp/call_log_manager/call_log_manager.cpp:56]: I wouldn't worry about that.

**Review by etienneptl — APPROVED** [review]
LGTM, excellent work. Just worried if the changes in `src/call_log/call_log.c` can have side effects. @bvpvamsikrishna, what do you think?

File: test/simulator/timestamp/call_log_manager/call_log_manager.h
Comment (etienneptl) [inline test/simulator/timestamp/call_log_manager/call_log_manager.h:32]: Can we add a comment to explain why this is required?

File: test/simulator/timestamp/src/include/session_simulator.h
Comment (bvpvamsikrishna) [reply test/simulator/timestamp/src/include/session_simulator.h:47]: I agree, that this part should be done in construction.

**Review by bvpvamsikrishna — APPROVED** [review]
I have one comment, lgtm after that comment is fixed.

File: test/simulator/timestamp/src/include/session_simulator.h
Comment (Sean04) [reply test/simulator/timestamp/src/include/session_simulator.h]: Fixed.

**Comment by Sean04** [comment]
I'm picking up this PR from @Siddhartha8899. After running the python tests I found that test_timestamp01 failed due the call long generator function incorrectly changing the return values in __session_commit_transaction in the case where the transaction had config string that was too large. I've now moved it so that it is only generated in the case that we are able to commit the transaction, however, we'll likely want to handle the error in a more robust way in the future.

File: test/simulator/timestamp/src/session_simulator.cpp
Comment (etienneptl) [inline test/simulator/timestamp/src/session_simulator.cpp:34]: You can use the init list:
```
session_simulator::session_simulator(): _txn_running(false) { }
```

File: test/simulator/timestamp/call_log_manager/call_log_manager.h
Comment (Sean04) [reply test/simulator/timestamp/call_log_manager/call_log_manager.h]: I've removed this since it should no longer be required.

**Review by etienneptl — APPROVED** [review]
LGTM!

**Pull request merged**
